src/Intervention/Image/AbstractDecoder.php

```diff
@@ -140,6 +140,10 @@ public function isGdResource()
             return (get_resource_type($this->data) == 'gd');
         }
 
+        if (is_object($this->data) && $this->data instanceOf \GdImage) {
```

Comment:
The `is_object` check is superfluous according to the [docs]
(https://www.php.net/manual/en/language.operators.type.php):

> instanceof does not throw any error if the variable being tested is not an object, it simply returns false.

I would also move this check to the top of the function as going forward, this is going to be the more common case.

Reply:
Thanks for pointing this out. The docs are indeed correct about the current behavior, but historically, `instanceof` required an object in PHP <= 7.2. 

- https://php.watch/versions/8.0/gdimage#gdimage-is-resource
- https://3v4l.org/YlFF5

I added that test because this library supports PHP 5.4 and later.

Reply:
That is just issue with literals, non-object variables work as far as PHP 5.0:

https://3v4l.org/32265

Reply:
You are right, I will push with the is_object() call removed.